src/units/unit.cpp

```diff
@@ -1121,6 +1116,7 @@ color_t unit::xp_color() const
 			color=amla_color;
 		}
 	}
+	}
```

Comment:
The controlled `if` in this `for` needs indentation.

Reply:
That's not the main issue with that code. What was previously a nested conditional with assignments, is now a loop over that same code with one of the conditionals altered.

It should instead be a loop (or better, a standard algorithm) checking whether the unit has an AMLA that is major, followed by an altered conditional.

There is no reason why the color assignment should be inside this loop.